tapdance/conn_dark_test.go

```diff
@@ -0,0 +1,75 @@
+package tapdance
+
+import (
+	"crypto/rand"
+	"fmt"
+	"testing"
+)
+
+func TestSelectIpv4(t *testing.T) {
+	fmt.Print("\nTesting IPv4 only' selection:\n")
+
+	_ddIpSelector, err := newDDIpSelector([]string{"192.122.190.0/24", "2001:48a8:687f:1::/64"}, false)
+	if err != nil {
+		t.Error("Failed IP selector initialization ", err)
+	}
+
+	fmt.Println("\nAvailable network blocks:")
+	for _, _net := range _ddIpSelector.nets {
+		fmt.Println(_net)
+		if _net.IP.To4() == nil {
+			t.Error("Encountered Non IPv4 Network block")
+			t.FailNow()
+		}
+	}
+
+	seed := make([]byte, 16)
+	_, err = rand.Read(seed)
+	if err != nil {
+		t.Errorf("Crypto/Rand error -- %s\n", err)
+		t.FailNow()
+	}
+
+	darDecoyIpAddr, err := _ddIpSelector.selectIpAddr(seed)
```

Comment:
I believe there were some intermittent errors with this function at some point. You can repeat this step with a new seed many many times to try to catch it.

Reply:
I cannot reproduce this. I believe this is an issue with the selection algorithm not with the v4/v6 update. We should probably open a separate issue.

Reply:
opened issue #19 to address this.